client/src/components/Instructions/index.js

```diff
@@ -92,6 +92,100 @@ export default class Instructions extends Component {
     );
   }
 
+  renderGaslessCounterSetup() {
+    const addressDefault = this.props.ganacheAccounts.length > 2 ? this.props.ganacheAccounts[2] : '<ADDRESS>';
+    return (
+      <div className={styles.instructions}>
+        <h2> Build your first app with ZepKit </h2>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            1. Ensure Docker is installed on your computer.
+          </div>
+          <div className={styles.code}>
+            <code>
+              <a href="https://docs.docker.com/install/">https://docs.docker.com/install/</a>
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            2. In another terminal, run the gas relay docker container with ganache, wait a few seconds for it to sping up (run inside your zepkit folder).
+          </div>
+          <div className={styles.code}>
+            <code>
+              npm explore tabookey-gasless npm run gsn-dock-relay-ganache
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            3. In a different terminal, init your project with ZeppelinOS (run inside your zepkit folder).
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos init zepkit
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            4. Add the GaslessCounter contract to your project (Note that a compilation warning will be thrown, you can ignore this).
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos add GaslessCounter
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            5. Connect with your local blockchain by opening a session (Note that we automatically prefilled your ganache account).
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos session --network development --from {addressDefault} --expires 3600
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            6. Let's compile the Gasless Counter contract.
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos push --deploy-dependencies
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            7. We create an instance of our contract and deploy it.
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos create GaslessCounter --init initialize
```

Comment:
As soon as I ran this command, the page reloaded automatically, and a new set of instructions was displayed (the one for removing the whitelist) - so I could no longer see the truffle migrate command.

Reply:
By the way, I had been running tests using the `zeppelin-solidity-hot-loader`, though I cleared it from my config before running this tutorial. If you cannot reproduce this, let me know so I clean my `node_modules` and try again.

Reply:
I was unable to reproduce this issue, could you also try the same steps with the regular counter?

Reply:
I re-run the tutorial on a new clone of the repo, and I ended up with the opposite problem: since no javascript source file is being modified during the steps, the reloader does not recompile and fails to pick up the newly deployed contract (stored in a json file), even after I reload the page. The Counter tutorial works around that by asking the user to add a `console.log` in App.js, maybe we could do the same here until we implement a more robust solution?

Reply:
I changed the last instruction to restart the client which will start with the deployed json files 👍